flink-table/flink-sql-parser/src/test/java/org/apache/flink/sql/parser/FlinkSqlParserImplTest.java

```diff
@@ -383,19 +383,21 @@ public void testCreateTableWithComplexType() {
 	}
 
 	@Test
-	public void testCreateTableWithDecimalType() {
+	public void testCreateTableWithComplexType() {
 		check("CREATE TABLE tbl1 (\n" +
-			"  a decimal, \n" +
-			"  b decimal(10, 0),\n" +
-			"  c decimal(38, 38),\n" +
+			"  a ARRAY<bigint>, \n" +
+			"  b MAP<int, varchar>,\n" +
+			"  c ROW<cc0 int, cc1 float, cc2 varchar>,\n" +
+			"  d MULTISET<varchar>,\n" +
 			"  PRIMARY KEY (a, b) \n" +
 			") with (\n" +
 			"  x = 'y', \n" +
 			"  asd = 'data'\n" +
 			")\n", "CREATE TABLE `TBL1` (\n" +
-			"  `A`  DECIMAL,\n" +
-			"  `B`  DECIMAL(10, 0),\n" +
-			"  `C`  DECIMAL(38, 38),\n" +
+			"  `A`  ARRAY< BIGINT >,\n" +
+			"  `B`  MAP< INTEGER, VARCHAR >,\n" +
+			"  `C`  ROW< `CC0` INTEGER, `CC1` FLOAT, `CC2` VARCHAR >,\n" +
+			"  `D`  MULTISET< VARCHAR >,\n" +
```

Comment:
`STRING` instead of `VARCHAR`? We should encourage this type everywhere (e.g. update also the DDL doc in table environments), because `VARCHAR` is equal to `VARCHAR(1)` in SQL standard.

Reply:
Actually for Calcite its default precision is 65536 which most of the sql vendors implement, i think `VARCHAR` type should be the first place although we introduce an alias `STRING` type, it is more widely used.

Reply:
According to the SQL standard section 6.1 clause (5):
"If <length> is omitted, then a <length> of 1 (one) is implicit." So we should encourage users to use `STRING` instead to prevent unwanted side effects.

Reply:
Also SQL Server has such behavior:
https://docs.microsoft.com/de-de/sql/t-sql/data-types/char-and-varchar-transact-sql?view=sql-server-2017
"When n isn't specified in a data definition or variable declaration statement, the default length is 1."

Reply:
But most of the users have used the `VARCHAR(65536)` for the long time, it would be a breaking change if we change it to default `VARCHAR(1)`, it is a SQL standard, but we do need follow everything.

Reply:
That's why I would suggest to introduce `STRING` as early as possible such that people start using it. We don't need to drop the old behavior immediately but can do it maybe for Flink 2.0. This should have been discussed in FLIP-37. Now it is documented and assumed at a couple of places.

Reply:
Jump in the discussion. Mysql forces to specify a length when using VARCHAR. So in order to avoid misuse `VARCHAR`, we can drop the support of `VARCHAR` (without length), and force users to specify a length. I think this doesn't break ANSI-SQL?

Reply:
Yes, that would also be an option. It is not really useful anyway and only exists to cover the corner cases of the standard.